### PR TITLE
[Bug](function) fix aggFnParams set not correct

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

fix core dump on #7288

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java [hunk 1]
```diff
@@ -222,7 +222,16 @@ protected FunctionCallExpr(FunctionCallExpr other) {
         } else {
             fnParams = new FunctionParams(other.fnParams.isDistinct(), children);
         }
-        aggFnParams = other.aggFnParams;
+
+        if (other.aggFnParams != null) {
+            if (other.aggFnParams.isStar()) {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java:227]: the code should be simple because is same as line 219

**Comment by yangzhg** [comment]
This bug will occur when enable_vec_exec, and your modification seems not fix this bug

**Review by yangzhg — DISMISSED** [review]
LGTM

**Review by yangzhg — DISMISSED** [review]
LGTM

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**